xilem/src/view/split.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::marker::PhantomData;
+
+use masonry::widgets::{self, Axis};
+use xilem_core::{DynMessage, View, ViewId, ViewMarker, ViewPathTracker};
+
+use crate::{Pod, ViewCtx, WidgetView};
+
+/// A container containing two other widgets, splitting the area either horizontally or vertically.
+///
+/// The split view will by default be draggable,
+/// which means that the relative size of each child view can be changed by dragging the mouse.
+///
+/// # Examples
+/// To create a split view, provide it with two child views.
+///
+/// ```ignore
+/// use xilem::view::{split, label};
+///
+/// split(
+///     label("Left view"),
+///     label("Right view")
+/// )
+/// ```
+///
+/// The split axis and split point can be changed.
+/// For the full list of modifiers see the [`Split`] struct.
+///
+/// ```ignore
+/// use xilem::view::{split, label};
+///
+/// split(label("Left view"), label("Right view"))
+///     .split_axis(Axis::Horizontal)
+///     .split_point(0.25)
+/// ```
+///
+pub fn split<State, Action, ChildA, ChildB>(
+    child1: ChildA,
+    child2: ChildB,
+) -> Split<ChildA, ChildB, State, Action>
+where
+    ChildA: WidgetView<State, Action>,
+    ChildB: WidgetView<State, Action>,
+{
+    Split {
+        split_axis: Axis::Horizontal,
+        split_point: 0.5,
+        min_size: (0.0, 0.0),
+        bar_size: 6.0,
+        min_bar_area: 6.0,
+        solid_bar: false,
+        draggable: true,
+        child1,
+        child2,
+        phantom: PhantomData,
+    }
+}
+
+/// The [`View`] created by [`split`].
+///
+/// See `split` documentation for more context.
+#[must_use = "View values do nothing unless provided to Xilem."]
+pub struct Split<ChildA, ChildB, State, Action = ()> {
+    split_axis: Axis,
+    split_point: f64,
+    min_size: (f64, f64), // Integers only
+    bar_size: f64,        // Integers only
+    min_bar_area: f64,    // Integers only
+    solid_bar: bool,
+    draggable: bool,
+    child1: ChildA,
+    child2: ChildB,
+    phantom: PhantomData<fn() -> (State, Action)>,
+}
+
+impl<ChildA, ChildB, State, Action> Split<ChildA, ChildB, State, Action> {
+    /// Set the split axis.
+    ///
+    /// Horizontal split axis means that the children are left and right.
+    /// Vertical split axis means that the children are up and down.
+    ///
+    /// The default split point is horizontal.
+    pub fn split_axis(mut self, axis: Axis) -> Self {
+        self.split_axis = axis;
+        self
+    }
+
+    /// Set the split point as a fraction of the split axis.
+    ///
+    /// The value must be between `0.0` and `1.0`, inclusive.
+    /// The default split point is `0.5`.
+    pub fn split_point(mut self, split_point: f64) -> Self {
+        assert!(
+            (0.0..=1.0).contains(&split_point),
+            "split_point must be in the range [0.0-1.0]!"
```

Comment:
```suggestion
            "split_point must be in the range [0.0-1.0], got {split_point}."
```

Is that a usual range syntax? I'd normally say `0.0..=1.0` or `[0.0, 1.0]`

Edit: Oh I see, this is retained from the previous... Hmm

Reply:
I changed it for both the View and the Widget 👍